osgar/lib/depth.py

```diff
@@ -58,13 +67,30 @@ def __init__(
         pzs = np.ones((self.camh, self.camw), dtype=np.float)
         # For each pixel in the image, a vector representing its corresponding
         # direction in the scene with a unit forward axis.
-        self.ps = np.dstack([pzs, pxs / fx, pys / fx])
+        self.ps = (self.camera_rot @ np.dstack([pzs, pxs / fx, pys / fx]).T.reshape((3, -1))).reshape((3, self.camw, self.camh)).T
 
 
 # Indices of directions in a matrix with 3D points.
 X, Y, Z = 0, 1, 2
 
 
+def rotation_matrix(yaw, pitch, roll):
+    cos_roll = np.cos(roll)
+    sin_roll = np.sin(roll)
+    roll_m = np.array([[1., 0., 0.], [0., cos_roll, -sin_roll],
+                       [0., sin_roll, cos_roll]])
+    cos_pitch = np.cos(pitch)
+    sin_pitch = np.sin(pitch)
+    pitch_m = np.array([[cos_pitch, 0, sin_pitch], [0, 1, 0],
+                        [-sin_pitch, 0, cos_pitch]])
+    cos_yaw = np.cos(yaw)
+    sin_yaw = np.sin(yaw)
+    yaw_m = np.array([[cos_yaw, -sin_yaw, 0], [sin_yaw, cos_yaw, 0],
+                      [0, 0, 1]])
+
+    return roll_m @ pitch_m @ yaw_m
+
+
```

Comment:
I think this does not really belong here. If we want to keep it here it should at least have a leading underscore so that someone does not start using it from somewhere else.

Reply:
Done.